### ISSUE TITLE
drush can't be found after lando destroy; lando start (but is installed in the container) 

### ISSUE THREAD

**Issue by ptmkenny** [lando/lando#1299]
Bug Report
==========

When I set up my site with Lando, everything was fine.  However, after I ran `lando destroy` and then rebuilt my pantheon drupal 8 app using a composer workflow (but no drush installed via composer in the repo), I am now unable to launch drush.  The error is:

```PHP Recoverable fatal error:  PHP Startup: Cannot set 'user' save handler by ini_set() or 
session_module_name() in Unknown on line 0
/app/vendor/drush/drush/drush.launcher: 3: cd: can't cd to ../drush/drush
/app/vendor/drush/drush/drush.launcher: 10: /app/vendor/drush/drush/drush.launcher: /drush.launcher: not found
```

It may be related to [this issue](
https://github.com/lando/lando/issues/931).

** Strange behavior**

When I ssh into `appserver`, the drush command gives me the same error inside `app` and subdirectories (working on a Pantheon repo with `app/web` as the site root).

However, if I leave the `app` directory (for example, `cd /`), then the drush command launches drush as expected.

I tried resetting Docker and downloading all the images again, but I'm still stuck with this error.

**Versions**

3.0.0-rc1 on Windows 10 Pro

**.lando.yml**

```yml
# If you change this file, lando rebuild.

# Change php and docroot in .pantheon.yml, not here.

name: MYSITE
recipe: pantheon
config:
  framework: drupal8
  site: MYSITE
  id: 123


# Additional mailhog proxy settings so we get a nice URL for the mailhog ui
proxy:
  mailhog:
    - mail.MYSITE.lndo.site
  pma:
    - pma.MYSITE.lndo.site

services:
  mailhog:
    # Use the latest mailhog version
    type: mailhog
    # List of services that we want to capture mail from
    # You can run `lando info` to get a list of services that are available
    # Mailhog expects that these services will be php services
    hogfrom:
      - appserver
    # Make sure port 1026 is open on your machine
    portforward: 1026
  database:
    portforward: 3306
  appserver:
    # xdebug config
    php: lando/php.ini
    build:
      - drush pm-enable drush_language -y
      # Add terminus plugins.
      - mkdir -p ~/.terminus/plugins
      - composer create-project -d ~/.terminus/plugins pantheon-systems/terminus-build-tools-plugin:~1
      - composer create-project --no-dev -d ~/.terminus/plugins pantheon-systems/terminus-drupal-console-plugin:~1
    extras:
      # Use custom settings.local.php; needs to be copied because we want a master under version control.
      - "cp /app/web/sites/default/lando.settings.local.php /app/web/sites/default/settings.local.php"
      # Enable twig debug and other dev services.
      - "cp /app/web/sites/default/twigdebug.services.yml /app/web/sites/default/services.yml"
    # overrides:
      # services:
        # ports: ['777:80']
  # Add a phpmyadmin db frontend
  pma:
    # Use the latest version of phpmyadmin
    type: phpmyadmin
    hosts:
      - database
    portforward: 1026
  # Create a node instance
  node:
    # Node: 8 is LTS, 9 is less stable
    type: node:8
    run:
      - "cd $LANDO_MOUNT && npm install"
      - "cd $LANDO_MOUNT && gulp"
    ssl: true
    globals:
      backstopjs: "latest"
      gulp-cli: "latest"
    # backstopjs example .lando.yml
    # https://gist.github.com/rachelwhitton/4254dabb04ad5b0e9c7fa11c07dc27be
    overrides:
      services:
        image: backstopjs/backstopjs

tooling:
  drush:
    service: appserver
    cmd:
      - /usr/local/bin/drush
      - "--root=/app/web"
  behat:
    service: appserver
    description: Run behat tests locally.
    cmd:
      - /app/vendor/bin/behat
  # chromedriver:
  #   service: chrome
  #   description: Run behat tests with chromedriver.
  phantomjs:
    service: appserver
    description: Run phantomjs commands.
  npm:
    service: node
  node:
    service: node
  gulp:
    service: node
  backstop:
    service: node


events:
  post-start:
    - appserver: cd $LANDO_MOUNT && composer install
  post-pull:
    - appserver: cd $LANDO_WEBROOT && cp ../lando/phpstorm_debug/phpinfo.php phpinfo.php
    - appserver: cd $LANDO_WEBROOT && cp ../lando/phpstorm_debug/test.php test.php
    # Ensure settings.local.php has been copied.
    - appserver: cd $LANDO_WEBROOT/sites/default && cp lando.settings.local.php settings.local.php
    # updatedb, config import, locale update (including custom language files), clear cache
    - appserver: cd $LANDO_WEBROOT && drush updatedb -y
    - appserver: cd $LANDO_WEBROOT && drush cr
    # Enable dev + admin modules
    - appserver: cd $LANDO_WEBROOT && drush config-split-import dev_config
    - appserver: cd $LANDO_WEBROOT && drush config-split-import admin_config
    # force dev_config admin_config config split to be active
    - appserver: cd $LANDO_WEBROOT && drush cset config_split.config_split.dev_config status TRUE
    - appserver: cd $LANDO_WEBROOT && drush cset config_split.config_split.admin_config status TRUE
    # GTM
    - appserver: cd $LANDO_WEBROOT && drush cset google_tag.settings include_environment TRUE -y
    - appserver: cd $LANDO_WEBROOT && drush cset google_tag.settings environment_id env-8 -y
```

**Tell us about the error you got**

```powershell
lando drush -- -vvv

verbose: Loading cli task mysql
verbose: Loading cli task node
verbose: Loading cli task npm
verbose: Loading cli task phantomjs
verbose: Loading cli task php
verbose: Loading cli task poweroff
verbose: Loading cli task pull
verbose: Loading cli task push
verbose: Loading cli task rebuild
verbose: Loading cli task redis-cli
verbose: Loading cli task restart
verbose: Loading cli task share
verbose: Loading cli task ssh
verbose: Loading cli task start
verbose: Loading cli task stop
verbose: Loading cli task switch
verbose: Loading cli task terminus
verbose: Loading cli task varnishadm
verbose: Loading cli task version
debug: Emitting event task-drush-answers
debug: Event task-drush-answers has 0 listeners
debug: CLI args _=[drush, -vvv], $0=C:\Program Files\Lando\bin\lando.exe C:\snapshot\lando\build\cli\bin\lando.js
debug: Emitting event task-drush-run
debug: Event task-drush-run has 0 listeners
debug: Emitting event pre-drush
debug: Event pre-drush has 0 listeners
debug: Emitting event pre-engine-up
debug: Event pre-engine-up has 0 listeners
debug: Trying to retrieve from file cache with key engineup
debug: File cache miss with key engineup
debug: Engine is up.
debug: Cached true with key engineup for {"persist":false,"ttl":5}
debug: Emitting event post-engine-up
debug: Event post-engine-up has 0 listeners
debug: Emitting event pre-engine-run
debug: Event pre-engine-run has 0 listeners
verbose: Running ["C:\\Program Files\\Docker\\Docker\\resources\\bin\\docker-compose.exe","--project-name","edoweb","--file","C:\\Users\\patri\\.lando\\compose\\edoweb\\edoweb-1.yml","exec","appserver","/bin/sh","-c","cd \"$LANDO_MOUNT\"&&/usr/local/bin/drush"] in C:\d\edoweb with options {"mode":"attach","detached":false}
debug: With stdio={"stdio":"inherit"}, mode=attach, detached=false
PHP Recoverable fatal error:  PHP Startup: Cannot set 'user' save handler by ini_set() or session_module_name() in Unknown on line 0
/app/vendor/drush/drush/drush.launcher: 3: cd: can't cd to ../drush/drush
/app/vendor/drush/drush/drush.launcher: 10: /app/vendor/drush/drush/drush.launcher: /drush.launcher: not foundverbose: C:\Program Files\Docker\Docker\resources\bin\docker-compose.exe,--project-name,edoweb,--file,C:\Users\patri\.lando\compose\edoweb\edoweb-1.yml,exec,appserver,/bin/sh,-c,cd "$LANDO_MOUNT"&&/usr/local/bin/drush finished with code 127
debug: Logging metrics data {"devMode":false,"nodeVersion":"v10.4.1","mode":"cli","os":{"type":"Windows_NT","platform":"win32","release":"10.0.17134","arch":"x64"},"product":"lando","version":"3.0.0-rc.1","message":"","stack":"Error\n    at module.exports.sh.Promise.try.then (C:\\snapshot\\lando\\build\\cli\\lib\\shell.js:0:0)\nFrom previous event:\n    at Shell.sh (C:\\snapshot\\lando\\build\\cli\\lib\\shell.js:0:0)\n    at dc (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\engine.js:0:0)\n    at compose (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\engine.js:0:0)\n    at exports.run.Promise.each.docker.isRunning.then.then (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\lib\\router.js:0:0)\nFrom previous event:\n    at exports.run.Promise.each.datum (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\lib\\router.js:0:0)\n    at runCallback (timers.js:696:18)\n    at tryOnImmediate (timers.js:667:5)\n    at processImmediate (timers.js:649:5)\nFrom previous event:\n    at Object.exports.run (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\lib\\router.js:0:0)\n    at module.exports.lando.run.data (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\engine.js:0:0)\n    at exports.engineCmd.daemon.up.then.then (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\lib\\router.js:0:0)\nFrom previous event:\n    at Object.exports.engineCmd (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\lib\\router.js:0:0)\n    at engineCmd (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\engine.js:0:0)\n    at run (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-engine\\engine.js:0:0)\n    at exports.runCommands.events.emit.then (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-tooling\\lib\\utils.js:0:0)\nFrom previous event:\n    at Object.exports.runCommands (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-tooling\\lib\\utils.js:0:0)\n    at run (C:\\snapshot\\lando\\build\\cli\\plugins\\lando-tooling\\tooling.js:0:0)\n    at module.exports.parseToYargs.events.emit.then.then.events.emit.then (C:\\snapshot\\lando\\build\\cli\\lib\\cli.js:0:0)\nFrom previous event:\n    at module.exports.parseToYargs.events.emit.then.then.answers (C:\\snapshot\\lando\\build\\cli\\lib\\cli.js:0:0)\n    at runCallback (timers.js:696:18)\n    at tryOnImmediate (timers.js:667:5)\n    at processImmediate (timers.js:649:5)\nFrom previous event:\n    at Object.handler (C:\\snapshot\\lando\\build\\cli\\lib\\cli.js:0:0)\n    at Object.module.exports.self.runCommand (C:\\snapshot\\lando\\build\\cli\\node_modules\\yargs\\lib\\command.js:170:22)\n    at parseArgs (C:\\snapshot\\lando\\build\\cli\\node_modules\\yargs\\yargs.js:920:28)\n    at Object.get [as argv] (C:\\snapshot\\lando\\build\\cli\\node_modules\\yargs\\yargs.js:860:16)\n    at bootstrap.then.Promise.try.then.then.then (C:\\snapshot\\lando\\build\\cli\\bin\\lando.js:0:0)\nFrom previous event:\n    at bootstrap.then.lando (C:\\snapshot\\lando\\build\\cli\\bin\\lando.js:0:0)\nFrom previous event:\n    at Object.<anonymous> (C:\\snapshot\\lando\\build\\cli\\bin\\lando.js:0:0)\n    at Module._compile (pkg/prelude/bootstrap.js:1252:22)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:711:10)\n    at Module.load (internal/modules/cjs/loader.js:610:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:549:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:541:3)\n    at Function.Module.runMain (pkg/prelude/bootstrap.js:1307:12)\n    at startup (internal/bootstrap/node.js:274:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:608:3)","action":"error","created":"2018-11-18T00:35:31.255Z"}

```

**Comment by ptmkenny** [comment]
Update: I still have the same issue (drush.launcher can't cd to ../drush/drush) after trying to reinstall with `lando destroy`, uninstalling lando, uninstalling docker, deleting the lando config directory, rebooting, and then installing everything again.

**Comment by markisatacomputer** [comment]
I experienced this problem too.  Recent reinstall of lando v3.0.0-rc.1 on mac 10.12.6.  I rebuilt Pantheon drupal 7 app, then experienced #1284 and this issue.  Fixed this issue with the following:

```
lando ssh

$: cd ~
$: wget https://github.com/drush-ops/drush/releases/download/8.1.18/drush.phar
$: mv drush.phar html/vendor/
$: chmod +x html/vendor/drush.phar
$: ln -s /var/www/html/vendor/drush.phar /usr/local/bin/drush
```

**Comment by kevinquillen** [comment]
This just happened to me as well.

I mistakenly tried `lando drush @site.alias cc all -y` with Docker not running, so Lando automatically init Docker (on Mac). It then attempted to execute the command and failed for some reason.

Restarting Lando had no effect. I `lando ssh` in and sure enough there is no Drush. I tried the steps in https://github.com/lando/lando/issues/1299#issuecomment-441844217, but my aliases were no longer recognized or copied in (per the steps outlined in my lando.yml).

My only recourse was to `lando destroy; lando start`, sync a database and continue working. I don't know why drush disappeared from the stack.

**Comment by pirog** [comment]
In ALMOST THERE RC2 we've changed a few things that should resolve this

1. Better PATH considerations for php containers 
https://docs.devwithlando.io/tutorials/php.html#path-considerations
2. Removal of drush launcher and simpler drush handling
https://docs.devwithlando.io/tutorials/drupal8.html#using-drush
3. Changed `lando restart` to not run any build steps

If you still experience this after upgrading to RC2 we highly recommend you submit a new ticket against that version
